lib/src/install/completion.rs

```diff
@@ -0,0 +1,298 @@
+    let install_config_kargs = install_config
+        .as_ref()
+        .and_then(|c| c.kargs.as_ref())
+        .into_iter()
+        .flatten()
+        .map(|s| s.as_str())
+        .collect::<Vec<_>>();
+    let kargsd = crate::kargs::get_kargs_in_root(deployment_root, std::env::consts::ARCH)?;
+    let kargsd = kargsd.iter().map(|s| s.as_str()).collect::<Vec<_>>();
+
+    current_kargs.append_argv(&install_config_kargs);
+    current_kargs.append_argv(&kargsd);
+    let new_kargs = current_kargs.to_string();
+    tracing::debug!("new_kargs={new_kargs}");
+
+    sysroot.deployment_set_kargs_in_place(deployment, Some(&new_kargs), cancellable)?;
+    Ok(())
+}
+
+/// A little helper struct which on drop renames a file. Used for putting back /etc/resolv.conf.
+#[must_use]
+struct Renamer<'d> {
+    dir: &'d Dir,
+    from: &'static Utf8Path,
+    to: &'static Utf8Path,
+}
+
+impl<'d> Renamer<'d> {
+    fn _impl_drop(&mut self) -> Result<()> {
+        self.dir
+            .rename(self.from, self.dir, self.to)
+            .map_err(Into::into)
+    }
+
+    fn consume(mut self) -> Result<()> {
+        self._impl_drop()
+    }
+}
+
+impl<'d> Drop for Renamer<'d> {
+    fn drop(&mut self) {
+        let _ = self._impl_drop();
+    }
+}
+/// Work around https://github.com/containers/buildah/issues/4242#issuecomment-2492480586
+/// among other things. We unconditionally replace the contents of `/etc/resolv.conf`
+/// in the target root with whatever the host uses (in Fedora 41+, that's systemd-resolved for Anaconda).
+#[context("Copying host resolv.conf")]
+fn ensure_resolvconf<'d>(rootfs: &'d Dir, proc1_root: &Dir) -> Result<Option<Renamer<'d>>> {
+    // Now check the state of etc/resolv.conf in the target root
+    let meta = rootfs
+        .symlink_metadata_optional(RESOLVCONF)
+        .context("stat")?;
+    let renamer = if meta.is_some() {
+        rootfs
+            .rename(RESOLVCONF, &rootfs, RESOLVCONF_ORIG)
+            .context("Renaming")?;
+        Some(Renamer {
+            dir: &rootfs,
+            from: RESOLVCONF_ORIG.into(),
+            to: RESOLVCONF.into(),
+        })
+    } else {
+        None
+    };
+    // If we got here, /etc/resolv.conf either didn't exist or we removed it.
+    // Copy the host data into it (note this will follow symlinks; e.g.
+    // Anaconda in Fedora 41+ defaults to systemd-resolved)
+    proc1_root
+        .copy(RESOLVCONF, rootfs, RESOLVCONF)
+        .context("Copying new resolv.conf")?;
+    Ok(renamer)
+}
+
+/// Bind a mount point from the host namespace into our root
+fn bind_from_host(
+    rootfs: &Dir,
+    src: impl AsRef<Utf8Path>,
+    target: impl AsRef<Utf8Path>,
+) -> Result<()> {
+    fn bind_from_host_impl(rootfs: &Dir, src: &Utf8Path, target: &Utf8Path) -> Result<()> {
+        rootfs.create_dir_all(target)?;
+        if rootfs.is_mountpoint(target)?.unwrap_or_default() {
+            return Ok(());
+        }
+        let target = format!("/mnt/sysroot/{target}");
```

Comment:
Maybe it's supposed to be obvious but what is `/mnt/sysroot`?

Reply:
EDIT: https://anaconda-installer.readthedocs.io/en/latest/mount-points.html#mnt-sysroot

Reply:
(Or better move it to a constant with a comment above it)